### PR TITLE
[proxima-core] Make sure AbstractWatermarkEstimator doesn't shift bac…

### PR DESCRIPTION
**Pull request by dmvk**
…k in time.

### CODE REVIEW

File: core/src/main/java/cz/o2/proxima/time/AbstractWatermarkEstimator.java [hunk 1]
```diff
@@ -26,6 +26,8 @@
   /** If estimator is in the idle state */
   private boolean isIdle = false;
 
+  private long lastWatermark = Long.MIN_VALUE;
```
Comment (LesTR) [inline core/src/main/java/cz/o2/proxima/time/AbstractWatermarkEstimator.java:29]: `Watermarks.MIN_WATERMARK`

File: beam/core/src/main/java/cz/o2/proxima/beam/direct/io/BeamCommitLogReader.java [hunk 2]
```diff
@@ -56,6 +56,7 @@
     private final DirectUnboundedSource source;
     @Getter private final BeamCommitLogReader reader;
 
+    private Instant previousWatermark = BoundedWindow.TIMESTAMP_MIN_VALUE;
```
Comment (LesTR) [inline beam/core/src/main/java/cz/o2/proxima/beam/direct/io/BeamCommitLogReader.java:59]: Maybe also here we should use `Watermarks.MIN_WATERMARK`

**Review by LesTR — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: beam/core/src/main/java/cz/o2/proxima/beam/direct/io/BeamCommitLogReader.java (hunk 2 above)
Comment (dmvk) [reply beam/core/src/main/java/cz/o2/proxima/beam/direct/io/BeamCommitLogReader.java:59]: this is a beam source, so we need a joda time instant here, so this is perfectly fine

File: core/src/main/java/cz/o2/proxima/time/AbstractWatermarkEstimator.java (hunk 1 above)
Comment (dmvk) [reply core/src/main/java/cz/o2/proxima/time/AbstractWatermarkEstimator.java:29]: 👍

File: beam/core/src/main/java/cz/o2/proxima/beam/direct/io/BeamCommitLogReader.java (hunk 2 above)
Comment (LesTR) [reply beam/core/src/main/java/cz/o2/proxima/beam/direct/io/BeamCommitLogReader.java:59]: true! My mistake.

**Review by LesTR — APPROVED** [review]

**Pull request merged**